app/src/main/java/com/summer/practice/tvtracker/TopRatedFragment.kt

```diff
@@ -13,6 +17,23 @@ class TopRatedFragment : Fragment() {
         savedInstanceState: Bundle?
     ): View? {
         // Inflate the layout for this fragment
+        MoviesRepository.getTopRatedList(
+            onSuccess = ::onPopularMoviesFetched,
+            onError = ::onError,
+        )
         return inflater.inflate(R.layout.fragment_top_rated, container, false)
     }
+
+    private fun onPopularMoviesFetched(movies: List<Movies>) {
+        val imageBaseUrl = "https://image.tmdb.org/t/p/w500"
+        val pathList =  movies.map {
+            it.copy(backdropPath = imageBaseUrl + it.backdropPath)
+        }
```

Comment:
duplicate code, let's extract the "data manipulation" into a single method that just returns to us a list where the full image url was set